### ISSUE TITLE
Widgets disappear when using DynamicWidgets and SortBy with virtual replicas

### ISSUE THREAD

**Issue by jsancho** [algolia/react-instantsearch#3564]
## 🐛 Bug description
I have an index for which I have configured some virtual replicas to allow sorting by attributes.

When I select a sort in the UI, the facet filters `<DynamicWidgets> <RefinementList/>...</DynamicWidgets>` disappear completely from the UI.

I've read the [sort by attribute](https://www.algolia.com/doc/guides/managing-results/refine-results/sorting/how-to/sort-by-attribute) docs a few times as well as the [in-depth replicas](https://www.algolia.com/doc/guides/managing-results/refine-results/sorting/in-depth/replicas/) ones.

But now I'm not entirely clear whether I'm missing some config, the behaviour is supported, or if there is a bug in the hooks library.

## 🔍 Bug reproduction

This happens on my private index, but can also be seen on this official example

https://codesandbox.io/s/fib6jm

If you select any sort by "price (asc)" or "price (desc)" you will see that all the dynamic widgets disappear from the left-hand side.

This issue does not seem to happen when using `react-instantsearch-dom` instead of the hooks library.
As you can see in this official demo.
https://react-instantsearch.netlify.app/examples/e-commerce

As far as I can tell both client apps are accessing the same index, so it could be down to the client library or its configuration.

Could you shed some light on what is required to combine relevant sort and facets when using the hooks library?

Thanks

## Environment

- React InstantSearch Hooks version: 6.31.0
- React version: 18.2.0
- Browser: Latest Chrome
- OS: Windows

**Comment by FabienMotte** [comment]
Hello @jsancho!

Thanks for your detailed report, the issue is related to the "Facet display" configuration on the Algolia dashboard.

When using `DynamicWidgets`, you need to configure the "Facet display" otherwise, no widgets will be displayed (except if you use the `fallbackComponent` option).

On the `instant_search` index, the "Facet display" configuration was properly set for the main index but empty for the virtual replica indices (`instant_search_price_asc` and `instant_search_price_desc`), it should be fixed now: https://codesandbox.io/s/fib6jm

You should be able to fix your issue by saving the "Facet display" settings on the dashboard and checking the options `Copy these settings to other indices and/or replicas` and `To all replicas of "your_index"`.

You can find more on [how to configure "Facet display" using the dashboard in this guide](https://www.algolia.com/doc/guides/building-search-ui/ui-and-ux-patterns/facet-display/js/#configuring-your-facet-display-using-the-dashboard).

Thank you!

**Comment by jsancho** [comment]
Hi @FabienMotte 

Thanks very much for the response and the additional context!
With your tips we've managed to get things working as we originally intended.

I'll leave a screenshot of the "copy settings" dialog in case somebody comes across this issue in the future.
![image](https://user-images.githubusercontent.com/1547884/179465864-99a934c3-3ea2-4ac8-8bcf-e1de696d77d1.png)

That options copies the facet ordering options to the virtual replicas and makes the widgets appear when choosing a sort on the UI.

The final fix has been amending our automation code that initialises the virtual replicas themselves. 
We now pass to the replica initialisation code the same configuration that we used for the main index. This achieves the same result without needing manual intervention post-deployment.

If I may, I would say that the documentation could be a bit more explicit on this front, to help guide new users who are not that familiar with the internals of Algolia.

In particular, there is a warning on the "facet display" configuration that mislead me down the wrong path.

<img width="783" alt="image" src="https://user-images.githubusercontent.com/1547884/179466918-5eafde78-ab04-4620-93dc-6b5accfdc6b5.png">

This message specifically
> A Virtual replica inherits this ettings from its primary index.
> If you want to change, do it on the primary directly.

Led me to think that I should somehow expect for the primary index settings to be re-used by all virtual replicas, instead of having to configure them individually.

Other than that, I'm really happy with the resolution and I'll close the issue as we got things sorted.

thanks again 👍
